src/formats/roam-json.ts

```diff
@@ -266,10 +357,56 @@ export class RoamJSONImporter extends FormatImporter {
 
 		if (json.children) {
 			for (const child of json.children) {
-				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true));
+				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true, checkYAMLoptions, false, '', this.oldestTimestamp, this.newestTimestamp));
+			}
+		}
+
+		// once processing children is completed, add the YAML to the top
+		// check if any YAML options are set
+		switch (!checkYAMLoptions) {
+			case this.fileDateYAML:
+			case this.titleYAML:
+			// add other options here
+			// case this.{option-name}:
+				checkYAMLoptions = true;
+		}
+
+		// add YAML frontmatter, if enabled
+		if (checkYAMLoptions === true && runYAMLoptions === true) {
+
+			let timeCreated = this.oldestTimestamp;
+
+			frontMatterYAML.push('---');
+
+			// if "add title" option enabled, quotes added to prevent errors in frontmatter
+			if (this.titleYAML === true) {
+				frontMatterYAML.push('title: "' + setTitleProperty +'"');
```

Comment:
I think the quotes would be much easier to read if we rewrote it like this:

```suggestion
				frontMatterYAML.push(`title: "${setTitleProperty}"`);
```

Reply:
agreed, thanks!